Paths added to messages and message cleanup

pboguslawski opened the pull request:
This mod adds path to error messages. Without it it's difficult for example
to find what path causes permission problems.

Related: https://help.nextcloud.com/t/ocp-files-notpermittedexception-could-not-create-folder/133818
Related: https://help.nextcloud.com/t/could-not-create-folder-false-error/65962/
Author-Change-Id: IB#1121253

CarlSchwan reviewed (approved):
Make sense to me, thanks!

kesselb reviewed (dismissed):
Hi, thanks for your pull request :+1: 

Please rebase this branch to master. Our usual process here is that every change goes to master and is backported (usually by a bot) if necessary.

pboguslawski committed:
Paths added to messages and message cleanup

This mod adds path to error messages. Without it it's difficult for example
to find what path causes permission problems.

Related: https://help.nextcloud.com/t/ocp-files-notpermittedexception-could-not-create-folder/133818
Related: https://help.nextcloud.com/t/could-not-create-folder-false-error/65962/
Author-Change-Id: IB#1121253
Signed-off-by: Pawel Boguslawski <pawel.boguslawski@ib.pl>

pboguslawski commented:
Just force pushed this branch with commit fcb4943071803716d8b7d51c973b440278f18c6d; this commit was created from master and DCO is present now - please change target branch of this PR from `stable24` to `master`.

szaimen commented:
/rebase

PVince81 commented on lib/private/Files/Node/Folder.php:
```diff
@@ -1,6 +1,7 @@
 <?php
 /**
  * @copyright Copyright (c) 2016, ownCloud, Inc.
+ * @copyright Copyright (c) 2022 Informatyka Boguslawski sp. z o.o. sp.k., http://www.ib.pl/
```
please don't change the copyright of the whole file

I guess maybe your IDE made this change ?

pboguslawski replied:
Removed copyright. We add it intentionally to fulfill the AGPL requirement 5a:
https://www.gnu.org/licenses/agpl-3.0.en.html

tcitworld replied:
You can use the `@author` doctype below instead

pboguslawski replied:
> You can use the @author doctype below instead

Please point to formal sentence that states that @author way is correct for AGPL. Seems that more standard/common is adding Copyright notice (vide [[1]](https://opensource.google/documentation/reference/copyright), [[2]](https://opensource.stackexchange.com/questions/8639/gpl-v3-0-is-there-a-standard-format-for-modification-notice)).

tcitworld replied:
I don't make such a claim, this was just a suggestion.

https://github.com/nextcloud/server/blob/master/contribute/HowToApplyALicense.md proves you right anyway.

PVince81 replied:
thanks, I wasn't aware of that practice as it seems very rare that people add a second copyright notice line.

then this is fine :+1:

PVince81 commented on lib/private/Files/Node/Folder.php:
```diff
@@ -1,6 +1,7 @@
 <?php
 /**
  * @copyright Copyright (c) 2016, ownCloud, Inc.
+ * @copyright Copyright (c) 2022 Informatyka Boguslawski sp. z o.o. sp.k., http://www.ib.pl/
```
```suggestion
 * @copyright Copyright (c) 2016, ownCloud, Inc.
```

pboguslawski committed:
Update lib/private/Files/Node/Folder.php

Co-authored-by: Vincent Petry <vincent@nextcloud.com>
Signed-off-by: Pawel Boguslawski <pawel.boguslawski@ib.pl>

pboguslawski committed:
Update Folder.php

Signed-off-by: Pawel Boguslawski <pawel.boguslawski@ib.pl>

pboguslawski committed:
Update Folder.php

Signed-off-by: Pawel Boguslawski <pawel.boguslawski@ib.pl>

PVince81 commented:
@pboguslawski do you might also adding a signed-off clause as per https://github.com/nextcloud/server/pull/33061/checks?check_run_id=7544272332 ? the DCO part is also stated in https://github.com/nextcloud/server/blob/master/contribute/HowToApplyALicense.md

pboguslawski commented:
DCO fixed.

ChristophWurst commented:
Unsure if failing CI is related or not. Could you please rebase to the latest `master` to have another clean CI run?

pboguslawski committed:
Merge master

pboguslawski commented:
Current master merged to this PR.

ChristophWurst commented:
Failing CI unrelated

come-nc commented:
/rebase
